### PR TITLE
Account Portal License Key - Activate & Manage Tests

### PR DESCRIPTION
**Pull request by Mitch-Budibase**
## Description
There are two test files -  license.activate.spec.ts and license.manage.spec.ts

These test files each contain a test:
- Creates, activates, and deletes an online license for a self hosted account - license.activate.spec.ts
- Retrieves plans, creates checkout session, and updates license - license.manage.spec.ts - WIP

Updated and created API files:
- StripeAPI
- LicenseAPI
- internal-api LicenseAPI
- index & AccountInternalAPI also updated to reflect API file changes

The license key manage test is a work in progress. The missing piece is 'Mimic checkout success' for Stripe. It is a requirement to create a customer using the Stripe SDK to finish this test

## Feature branch env
[Feature Branch Link](http://fb-account-portal-api-license-key.fb.qa.budibase.net)

### CODE REVIEW

File: qa-core/src/account-api/tests/licensing/license.activate.spec.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    )
+
+    // Activate license key
+    await config.internalApi.license.activateLicenseKey({ licenseKey })
+
+    // Verify license updated with new feature
+    await config.doInNewState(async () => {
+      await config.loginAsAccount(createAccountRequest)
+      const [selfRes, body] = await config.api.accounts.self()
+      expect(body.license.features[0]).toBe("appBackups")
```
Comment (shogunpurple) [inline qa-core/src/account-api/tests/licensing/license.activate.spec.ts:59]: I would probably do a more comprehensive test here - like test the shape of the full JSON

**Review by shogunpurple — APPROVED** [review]
LGTM! Just a few comments

File: qa-core/src/account-api/tests/licensing/license.activate.spec.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          features: [Feature.APP_BACKUPS],
+        },
+      }
+    )
+
+    // Activate license key
+    await config.internalApi.license.activateLicenseKey({ licenseKey })
+
+    // Verify license updated with new feature
+    await config.doInNewState(async () => {
```
[the remark this answers is not in the document]
Comment (Mitch-Budibase) [reply qa-core/src/account-api/tests/licensing/license.activate.spec.ts:56]: An account is created at the start of each test run. If another account is created during the test and needs to be accessed, we can use `doInNewState` e.g. when wanting to login to the account

[the remark this answers is not in the document]
Comment (shogunpurple) [reply qa-core/src/account-api/tests/licensing/license.activate.spec.ts:56]: gotcha - that's a handy helper then!

**Pull request merged**